RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -175,48 +359,242 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
+
+        // fill the zero energy cells in 1D theta-E profile
+        for (int i = result.first.front(); i <= result.first.back(); i += nMergedThetaCells[layer+startPositionToFill]) {
+          if (std::find(result.first.begin(), result.first.end(), i) == result.first.end()) {
+            auto it = std::lower_bound(result.first.begin(), result.first.end(), i);
+            int idx = it - result.first.begin();
+            result.first.insert(it, i);
+            result.second.insert(result.second.begin() + idx, 0);
+          }
+        }
+        theta_E_pair.push_back(result);
+
+        // loop over theta IDs to find the local E maxima
+        for (size_t i = 0; i < theta_E_pair[layer+startPositionToFill].second.size(); i ++) {
+          //std::cout << i << " " << theta_E_pair[layer+startPositionToFill].first[i] << " " << theta_E_pair[layer+startPositionToFill].second[i] << std::endl;
+          if (
+            (i == 0 && theta_E_pair[layer+startPositionToFill].second[i] > theta_E_pair[layer+startPositionToFill].second[i+1]) ||
+            (i == theta_E_pair[layer+startPositionToFill].second.size()-1 && theta_E_pair[layer+startPositionToFill].second[i] > theta_E_pair[layer+startPositionToFill].second[i-1]) ||
+            (i != 0 && i != (theta_E_pair[layer+startPositionToFill].second.size()-1) &&
+              theta_E_pair[layer+startPositionToFill].second[i] > theta_E_pair[layer+startPositionToFill].second[i-1] &&
+              theta_E_pair[layer+startPositionToFill].second[i] > theta_E_pair[layer+startPositionToFill].second[i+1])
+           ) {
+            local_E_Max[layer+startPositionToFill].push_back(theta_E_pair[layer+startPositionToFill].second[i]);
+            local_E_Max_theta[layer+startPositionToFill].push_back(theta_E_pair[layer+startPositionToFill].first[i]);
+          }
+        }  // end of loop over theta IDs
+
+        if (local_E_Max[layer+startPositionToFill].empty()) {
+          E_cell_Max[layer+startPositionToFill] = 0.;
+          E_cell_secMax[layer+startPositionToFill] = 0.;
+          E_cell_Max_theta[layer+startPositionToFill] = 0.;
+          E_cell_secMax_theta[layer+startPositionToFill] = 0.;
+          E_cell_Min[layer+startPositionToFill] = 0.;
+        } else if (local_E_Max[layer+startPositionToFill].size() < 2) {
+          E_cell_Max[layer+startPositionToFill] = local_E_Max[layer+startPositionToFill][0];
+          E_cell_secMax[layer+startPositionToFill] = 0.;
+          E_cell_Max_theta[layer+startPositionToFill] = local_E_Max_theta[layer+startPositionToFill][0];
+          E_cell_secMax_theta[layer+startPositionToFill] = local_E_Max_theta[layer+startPositionToFill][0];
+          E_cell_Min[layer+startPositionToFill] = 0.;
+        } else {
+          std::vector<double> sortedVec = local_E_Max[layer+startPositionToFill];
+          // move the top 2 max to the beginning
+          std::partial_sort(sortedVec.begin(), sortedVec.begin()+2, sortedVec.end(), std::greater<double>());
```

Comment:
if you want only the first two, rather than the first three, you could  actually do
```suggestion
          std::partial_sort(sortedVec.begin(), sortedVec.begin()+1, sortedVec.end(), std::greater<double>());
```